flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/split/assigner/UnboundedSplitAssigner.java

```diff
@@ -0,0 +1,195 @@
+                                                                                    .compareTo(
+                                                                                            oldestPartitionId)
+                                                                            >= 0)
+                                                    .orElse(true))
+                            .filter(pIdStatus -> pIdStatus.isCompleted())
+                            .filter(
+                                    pIdStatus ->
+                                            !this.lastSeenPartitions.contains(
+                                                    pIdStatus.getPartitionId()))
+                            .collect(Collectors.toList());
+            return new DiscoveryResult(
+                    newPartitions.stream()
+                            .map(p -> p.getPartitionId())
+                            .collect(Collectors.toList()),
+                    newPartitions.stream()
+                            .map(
+                                    pId ->
+                                            formatPartitionRestrictionBasedOnInfo(
+                                                    Optional.of(pId.getInfo()),
+                                                    pId.getInfo().getColumnName(),
+                                                    partitionValuesFromIdAndDataType(
+                                                                    Lists.newArrayList(
+                                                                            pId.getPartitionId()),
+                                                                    pId.getInfo().getColumnType())
+                                                            .get(0)))
+                            .flatMap(
+                                    restriction ->
+                                            SplitDiscoverer.discoverSplits(
+                                                    options,
+                                                    DataFormat.AVRO,
+                                                    this.readOptions.getColumnNames(),
+                                                    shouldAppendRestriction(
+                                                            this.readOptions.getRowRestriction(),
+                                                            restriction),
+                                                    this.readOptions.getSnapshotTimestampInMillis(),
+                                                    this.readOptions.getMaxStreamCount())
+                                                    .stream())
+                            .collect(Collectors.toList()));
+        } catch (Exception ex) {
+            throw new RuntimeException("Problems while trying to discover new splits.", ex);
+        }
+    }
+
+    @VisibleForTesting
+    String shouldAppendRestriction(String existing, String newRestriction) {
```

Comment:
We should name it combineRestriction or something since it's not returning a boolean.

Reply:
Fixed. Thanks.